### DISCUSSION TITLE
Building and running SysON in Ubuntu 22 - instructions

### DISCUSSION THREAD

**Discussion by petrotta** [eclipse-syson/syson#47] [General]
I used the following script to pull the source, compile the front/backend, and deploy using docker.  There is a youtube video that describes installation on Windows, but I have found deployment on linux to be more reliable.  I tested this on Ubuntu 22.

After the script runs, try http://localhost:8080/

~~~
#!/bin/bsh

# ****************************************************************************************************
# Step 1: get source code
# Update the git project, or clone it if already exists
if cd syson;  then  git pull; 
else git clone https://github.com/eclipse-syson/syson; cd syson;  fi


# Github login information, required to pull packages
# To download and install packages from a repository, your personal access token (classic) MUST have the read:packages scope
# see https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages
export USER=<github user name>
export PASSWORD=<use github personal access token WITH read:packages scope, see comments above> 
export EMAIL=<your email>


# ****************************************************************************************************
# Step 2: Front end -- Configure build environment and build
# use nvm to select the correct version of node.js, which is 18.7
nvm use 18.7

npm-cli-login -u $USER -p $PASSWORD -e $EMAIL  -s @NAMESPACE -r https://npm.pkg.github.com

npm ci
npx turbo run build


# post processing
mkdir -p backend/application/syson-frontend/src/main/resources/static
cp -R frontend/syson/dist/* backend/application/syson-frontend/src/main/resources/static

# ****************************************************************************************************
# Step 3: Back end

mvn -U -B -e clean verify --settings settings.xml

# ****************************************************************************************************
# Step 4: Build docker image and run

# Create the docker image:
sudo docker build -t syson backend/application/syson-application

# Run the app:
cd backend/application/syson-application

sudo docker-compose up

~~~

**Comment by ziruili-tu-ilmenau** [comment]
Hi Petrotta,
many thanks for this instruction. My name is Zirui Li and I'm trying to deploy SysON on my Windows.
First I have tried the WSL2 on Windows and follow your script, but there are some errors.
I have seen that you mentioned about a youtube video that describes installation on Windows, but unfortunatly I didn't find it. Could you please share the link also? Or give me the keywords for the video?

Thanks.
Zirui Li

**Reply by AxelRICHARD** [reply]
Hello, here is the link to the video: https://www.youtube.com/watch?v=IgFEfofNCbM
An official documentation with build & installation instructions will be available soon.
Regards,

**Comment by TonyCornuaud** [comment]
Hello,

I followed your instructions and I succeed to install SysON on a Raspberry Pi Model 4 8Go with Ubuntu 22.04. It works ! Sure with latency ;-) But it permits to have a first look of the tool about graphical notation of SysML v2.

![Screenshot from 2024-02-10 11-26-33](https://github.com/eclipse-syson/syson/assets/46522476/572f1726-5819-4fcb-873c-fd4dd07c2874)


I have one question...

I would like to know when the textual notation would be available inside SysON tool?
For the moment, I'm using Jupiter Lab.

Thank you,
Tony.

**Reply by AxelRICHARD** [reply]
Hello Tony,

Nice to see SysON running on a Raspberry!
Yes, we plan to have interoperability with the textual notation in 2024.
You can find the information and others about initial roadmap in the https://projects.eclipse.org/projects/modeling.syson/reviews/creation-review.

You can also find further information in https://projects.eclipse.org/projects/modeling.syson: 

> Additionally, Eclipse SysON embraces the standard API for interconnection, enhancing the interoperability of these vital modeling resources and will support the SysML v2 textual specifications as an exchange format, to ensure seamless transitions. 

Regards,

**Comment by roboviv** [comment]
Hello, where did you find the instructions to install SysON? Pardon my ignorance but I could not find on GitHub readme or website.I just need to try SysON on a Windows PC.I am about to install Sirius Web per their instructions but do not know how to add SysON.Thank you
-------- Original message --------From: TonyCornuaud ***@***.***> Date: 2/13/24  12:57 PM  (GMT-08:00) To: eclipse-syson/syson ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [eclipse-syson/syson] Building and running SysON in Ubuntu 22 -
  instructions (Discussion #47) 
Hello,
I followed your instructions and I succeed do install SysON on a Raspberry Pi Model 4 8Go with Ubuntu 22.04. It works ! Sure with latency ;-) But it permits ti have a first look of the tool about graphical notation of SysML v2.
Screenshot.from.2024-02-10.11-26-33.png (view on web)
I have one question...
I would like to know when the textual notation would be available inside SysON tool?
For the moment, I'm using Jupiter Lab.
Thank you,
Tony.

**Comment by AxelRICHARD** [comment]
Hello,
A youtube video has been made few weeks ago by an someone outside from the team: https://www.youtube.com/watch?v=IgFEfofNCbM
An official documentation (made by the team) with build & installation instructions will be available soon.
Regards,
